src/include/serial_inline.h

```diff
@@ -282,7 +282,13 @@ __wt_update_serial(WT_SESSION_IMPL *session, WT_CURSOR_BTREE *cbt, WT_PAGE *page
         obsolete_timestamp = page->modify->obsolete_check_timestamp;
         if (!__wt_txn_visible_all(session, txn, obsolete_timestamp)) {
             /* Try to move the oldest ID forward and re-check. */
-            WT_RET(__wt_txn_update_oldest(session, 0));
+            ret = __wt_txn_update_oldest(session, 0);
+            /*
+             * We cannot proceed if we fail here as we have inserted the updates to the update
+             * chain. Panic instead.
```

Comment:
I guess this is reasonable, though to be honest, I don't think updating oldest is necessary for correctness. I read through the code and don't believe it can actually return an error. The only functions that return errors that are called by `txn_update_oldest` are `wt_try_readlock` and `wt_try_writelock`. The only error codes those functions ever return are `EBUSY` - and that `EBUSY` is consumed by the `txn_update_oldest` function and success is returned.
If update oldest could actually generate an error, we'd be better to log it here and proceed rather than panic, but as it stands I'm OK with this change.

Reply:
That was a long comment - would you mind extending the comment to explain that a non-zero error isn't every actually generated at the moment, if the code is changed in the future to generate errors we should consider handling this differently than via a panic?

Reply:
Yes, sure

Reply:
Fix in a281d1815d87ef96190ca60fce2733a81f133471